src/PolygonDraw/reducer.ts

```diff
@@ -32,6 +33,14 @@ export const polygonEditReducer = (state: PolygonEditState, action: Actions): Po
                 polygons: [...action.payload]
             };
         }
+        case SET_POLYGON: {
+            const newPolygons = state.polygons.slice();
```

Comment:
I think the polygon with the active index should be replaced

Reply:
That's what's happening in the next line. `.slice()` is just a way of creating a copy of the array.